git/config.go

```diff
@@ -140,31 +140,30 @@ func (c *Configuration) Sources(dir string, optionalFilename string) ([]*Configu
 	if err != nil {
 		return nil, err
 	}
+	configs := make([]*ConfigurationSource, 0, 2)
 
 	bare, err := IsBare()
-	if err != nil {
-		return nil, err
-	}
-
-	// First try to read from the working directory and then the index if
-	// the file is missing from the working directory.
-	var fileconfig *ConfigurationSource
-	if !bare {
-		fileconfig, err = c.FileSource(filepath.Join(dir, optionalFilename))
-		if err != nil {
-			if !os.IsNotExist(err) {
-				return nil, err
+	if err == nil {
+
```

Comment:
Minor note: I think there's an extra line of whitespace here.

I suppose one could avoid indenting this block by instead doing something like:
```go
	if err != nil {
		return append(configs, gitconfig), nil
	}
```
But this approach has the advantage of not repeating the `return` line, so I think it's actually better as written.

Reply:
Whitespace removed.  And I agree this is slightly nicer to avoid duplication.